go/test/endtoend/vtorc/api/api_test.go

```diff
@@ -107,6 +107,11 @@ func TestProblemsAPI(t *testing.T) {
 		assert.Equal(t, 200, status, resp)
 		assert.Contains(t, resp, fmt.Sprintf(`"Port": %d`, replica.MySQLPort))
 
+		// Verify that filtering by keyspace also works in the API as intended
+		status, resp = utils.MakeAPICall(t, vtorc, "/api/replication-analysis?keyspace=ks")
+		assert.Equal(t, 200, status, resp)
+		assert.Contains(t, resp, fmt.Sprintf(`"Port": %d`, replica.MySQLPort))
+
 		// Check that filtering using keyspace and shard works
```

Comment:
I didn't add a shard query because that isn't a use-case we care for. It doesn't make any sense to filter only by shard and not the keyspace. Why would a user want the information regarding multiple shards with the same name from different keyspaces?

Reply:
However, if you think that these tests are worthwhile to add, I don't mind.

Reply:
Ideally if someone provided a query with just shard and no keyspace, we should return an error.

Reply:
Fixed and added tests for it.